### ISSUE TITLE
Any plans on extending the PKI backend?

### ISSUE THREAD

**Issue by ttaveira** [spring-projects/spring-vault#125]
Hi,

I see the current PKI backend implementation supports only one operation - [generate certificate](https://www.vaultproject.io/api/secret/pki/index.html#generate-certificate).
Are there any plans to extend it to support more operations?

Thank you,
Tiago

**Comment by mp911de** [comment]
Until now, no one asked for more features. Certificate generation is the most common use case.  Because transformation into a `KeyStore` is quite cumbersome we provided some additional API to perform the actual transformations. 

What functionality are you looking for? Can you achieve it with `VaultTemplate`'s read/write/delete operations?

**Comment by ttaveira** [comment]
Was mostly looking at [Read CRL](https://www.vaultproject.io/api/secret/pki/index.html#read-crl), [Sign Certificate](https://www.vaultproject.io/api/secret/pki/index.html#sign-certificate), and [Revoke Certificate](https://www.vaultproject.io/api/secret/pki/index.html#revoke-certificate).

Can you elaborate on the KeyStore transformation API? Are you talking about KeystoreUtil and SslConfiguration?
Basically I'm exploring the use of certificates obtained through the PKI backend for mutual authentication processes in Spring Boot.

**Comment by mp911de** [comment]
We provide [`CertificateBundle`](http://docs.spring.io/spring-vault/docs/current/api/org/springframework/vault/support/CertificateBundle.html) as an entry point to obtain private keys and certificates as `KeySpec` respective `X509Certificate` objects. 

It makes sense to provide a similar data structure for certificate signing and an API to create a `KeyStore` (trust store) from the signed certificate.

How would you use the CRL? Download it and distribute the CRL to an HTTP server? Or do you intend to DER-decode the CRL and implement a `TrustManager` yourself?

**Comment by ttaveira** [comment]
Hi @mp911de, just wanted to thank you on the above enhancement!
